File: lib/Models/Playlist.dart
class Playlist {
  int id;
  String name;
  String? artist; // Thêm trường artist
  String? image;
  List<int> songIds; // Danh sách ID bài hát

  Playlist({
    required this.id,
    required this.name,
    this.artist, // Thay đổi trường artist thành optional
    this.image,
    required this.songIds,
  });

  factory Playlist.fromJson(Map<String, dynamic> json) {
    print(json); // In ra dữ liệu JSON để kiểm tra
    return Playlist(
      id: json['id'] ?? 0,
      name: json['name'] ?? 'Tên Playlist Mặc Định', // Gán mặc định cho name
      artist: json['artist'] ?? 'Nghệ Sĩ Mặc Định', // Gán mặc định cho artist
      image: json['image'],
      songIds: List<int>.from(json['songs']?.map((song) => song['id']) ?? []), // Kiểm tra trường 'songs'
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'artist': artist, // Thêm trường artist vào JSON
      'image': image,
      'songIds': songIds,
    };
  }
}

File: src/main/java/app_music/demo/Model/Playlist.java
package app_music.demo.Model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Playlist {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Tên danh sách phát là bắt buộc!")
    private String name;

    private String image; // Đường dẫn hình ảnh của playlist

    @NotBlank(message = "Nghệ sĩ là bắt buộc!") // Bắt buộc nhập nghệ sĩ
    private String artist; // Nghệ sĩ của playlist

    @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    @JoinTable(
            name = "playlist_songs",
            joinColumns = @JoinColumn(name = "playlist_id"),
            inverseJoinColumns = @JoinColumn(name = "song_id")
    )
    private List<Song> songs = new ArrayList<>(); // Danh sách bài hát trong playlist
}